### ISSUE TITLE
Crash if using -x and a connection is made before a process binds

### ISSUE THREAD

**Issue by Darkhogg** [strongloop/node-foreman#69]
I run my app locally using the following command:

    nf start -x 4000 -j Procfile.local

If a connection is made during startup, when no process is bound to its port, the following exception is thrown:

```
/usr/lib/node_modules/foreman/node_modules/http-proxy/lib/http-proxy/index.js:119
    throw err;
          ^
Error: connect ECONNREFUSED
    at exports._errnoException (util.js:746:11)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1000:19)
```

After that, node-foreman immediately quits.

I have my processes run using `nodemon` so that changes are reflected immediately in the app, so I have to wait a few moments to even try to make a request so that it doesn't crash.

Is there a solution to this from my part? If not, a fix would be nice, even if it's just refusing the connection and continue, so that node-forman doesn't crash.

**Comment by rmg** [comment]
This may have been fixed by #97 which has been merged and included in the latest pre-release, [2.0.0-1](https://github.com/strongloop/node-foreman/releases/tag/v2.0.0-1). You can install it using `npm -g i foreman@next`.

@Darkhogg I realize this issue is over a year old now, but just in case.. if you are able to confirm that the fix works for you, can you close this issue?

**Comment by Darkhogg** [comment]
I just tested it and it's fixed. Instead of crashing, now it prints the following ugly error on the console _twice_:

```
Proxy Error:  { Error: connect ECONNREFUSED 127.0.0.1:5000
    at Object.exports._errnoException (util.js:896:11)
    at exports._exceptionWithHostPort (util.js:919:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1073:14)
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 5000 }
```

In the browser I just see `Upstream Proxy Error`.

It would be really nice if the logs didn't get that ugly message (*twice!*) but all things considered, `nf` doesn't crash anymore so I'm closing this.

Thanks!
